src/flint/types/fmpz_mod_mpoly.pyx

```diff
@@ -890,7 +909,7 @@ cdef class fmpz_mod_mpoly(flint_mpoly):
             c = fmpz.__new__(fmpz)
             fmpz_set((<fmpz>c).val, &fac.exp[i])
 
-            res[i] = (u, c)
+            res[i] = (u, int(c))
```

Comment:
We could likely get away with using `fmpz_get_ui` here although it would be fiddly to do correctly.

Probably not worth it.